### PR TITLE
datasource: new AgentPool Data Source

### PR DESCRIPTION
**Pull request by by-nelson**
### About
Greetings, I'm Nelson and I work for Joby Aviation, we're interested on using Terraform and this plugin to manage our top level configuration and resources for TeamCity like the _Root project, Agent Pools and Cloud Profiles. I'm more than happy to contribute to this great project, I have a little experience after having to create custom tools that consume the TeamCity Rest API to generate reports, custom metrics for Prometheus and more. 

In this PR I'm adding a data source to get an Agent Pool by name as the first step to have an Agent Pool resource. Please let me know your thoughts and any changes needed for an approval.

> This is a simple data source but I expect to give it a bigger scope (get data about projects and agents associated with the pool) based on your feedback and our team needs

### Changelog

1. Added local teamcity service on Docker-compose that uses the public registry (I don't have access to the one that exists already in compose)
2. Added Agent Pool Data Source that gets an Agent Pool Id and Agents size
3. Added client function to get a resource and unmarshal the response in the same unit
4. Added client function to verify the connection to the server
5. Calling the verify connection function when doing the provider configuration
6. Added example 8 for the usage of the data source

### Output

```sh
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - jetbrains/teamcity in /home/nelson.alvarez/go/bin
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
var.teamcity_token
  Enter a value: 

data.teamcity_pool.default: Reading...
data.teamcity_pool.test_pool: Reading...
data.teamcity_pool.test_pool: Read complete after 0s [name=TestPool]
data.teamcity_pool.default: Read complete after 0s [name=Default]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

default = {
  "id" = 0
  "name" = "Default"
  "size" = 0
}
test = {
  "id" = 1
  "name" = "TestPool"
  "size" = 28
}

```

### CODE REVIEW

**Comment by ChildOfJustice** [comment]
Hello, Nelson!
Thank you very much for your contribution! We looked at the code and really appreciate your work, everything looks great except one change in the client.go - we would like to ask you to use our current approach for handling http requests.
I left a comment about this in the review, we are going to change this client and it would be nice to keep all current files and have the same approach so its easier to refactor.

Thanks again for the changes, especially for the VerifyConnection() - we really needed that!

File: teamcity/pool/pool-data-source.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package pool
```
Comment (ChildOfJustice) [inline teamcity/pool/pool-data-source.go:1]: May I also please ask you not to put this into a separate package, lets leave it in the flat directory structure as everything else

File: client/client.go [hunk 2]
```diff
@@ -161,3 +164,71 @@ type Property struct {
...
+	}
+
+	// Unmarshal the response
+	err = json.Unmarshal(response.Body, resp)
+	if err != nil {
+		return Response{}, err
+	}
+
+	return response, nil
+}
```
[the remark this answers is not in the document]
Comment (by-nelson) [reply client/client.go]: Should be using the current approach now, thank you for the review!

File: teamcity/pool/pool-data-source.go (hunk 1 above)
Comment (ChildOfJustice) [reply teamcity/pool/pool-data-source.go]: Thanks! No, I agree with your idea to separate our data representation to a separate package, will also try to properly refactor the rest of the code to do the same

**Pull request merged**